github: https://github.com/MaharshSuryawala/Predict-Disease-From-Symptoms/blob/master/final_disease_symptom_data.csv

## Libiraries

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.preprocessing import MinMaxScaler, StandardScaler
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.tree import DecisionTreeClassifier

**Read the dataset**

In [2]:
df = pd.read_csv('SymptomsData.csv')

display the dataset

In [3]:
df

diseases  Heberden's node  Murphy's sign  \
0         hypertensive  disease                0              0   
1                      diabetes                0              0   
2            depression  mental                0              0   
3           depressive disorder                0              0   
4    coronary  arteriosclerosis                0              0   
..                          ...              ...            ...   
143                       ileus                0              0   
144                    adhesion                0              0   
145                    delusion                0              0   
146              affect  labile                0              0   
147            decubitus  ulcer                0              0   

     Stahli's line  abdomen acute  abdominal bloating  abdominal tenderness  \
0                0              0                   0                     0   
1                0              0                   0                     0   
2                0              0                   0                     0   
3                0              0                   0                     0   
4                0              0                   0                     0   
..             ...            ...                 ...                   ...   
143              0              1                   0                     1   
144              0              0                   0                     0   
145              0              0                   0                     0   
146              0              0                   0                     0   
147              0              0                   0                     0   

     abnormal sensation  abnormally  hard consistency  \
0                     0                             0   
1                     0                             0   
2                     0                             0   
3                     0                             0   
4                     0                             0   
..                  ...                           ...   
143                   0                             0   
144                   0                             0   
145                   0                             0   
146                   0                             0   
147                   0                             0   

     abnormally hard consistency  ...  vomiting  weepiness  weight gain  welt  \
0                              0  ...         0          0            0     0   
1                              0  ...         1          0            0     0   
2                              0  ...         0          1            0     0   
3                              0  ...         0          1            0     0   
4                              0  ...         0          0            0     0   
..                           ...  ...       ...        ...          ...   ...   
143                            0  ...         0          0            0     0   
144                            0  ...         1          0            0     0   
145                            0  ...         0          1            0     0   
146                            0  ...         0          0            0     0   
147                            0  ...         0          0            0     0   

     wheelchair bound  wheezing  withdraw  worry  yellow  sputum  \
0                   0         0         0      0               0   
1                   0         0         0      0               0   
2                   0         0         0      1               0   
3                   0         0         0      1               0   
4                   0         0         0      0               0   
..                ...       ...       ...    ...             ...   
143                 0         0         0      0               0   
144                 0         0         0      0               0   

**Check null values**

In [4]:
(df.isnull().sum()).sum()

0

Encoding target values

In [5]:
# Import label encoder 
from sklearn import preprocessing
# label_encoder object knows how to understand word labels. 
label_encoder = preprocessing.LabelEncoder()
# Encode labels in column 'diseases'. 
df['diseases']= label_encoder.fit_transform(df['diseases']) 
df

diseases  Heberden's node  Murphy's sign  Stahli's line  abdomen acute  \
0          80                0              0              0              0   
1          47                0              0              0              0   
2          45                0              0              0              0   
3          46                0              0              0              0   
4          34                0              0              0              0   
..        ...              ...            ...            ...            ...   
143        83                0              0              0              1   
144         6                0              0              0              0   
145        42                0              0              0              0   
146         7                0              0              0              0   
147        36                0              0              0              0   

     abdominal bloating  abdominal tenderness  abnormal sensation  \
0                     0                     0                   0   
1                     0                     0                   0   
2                     0                     0                   0   
3                     0                     0                   0   
4                     0                     0                   0   
..                  ...                   ...                 ...   
143                   0                     1                   0   
144                   0                     0                   0   
145                   0                     0                   0   
146                   0                     0                   0   
147                   0                     0                   0   

     abnormally  hard consistency  abnormally hard consistency  ...  vomiting  \
0                               0                            0  ...         0   
1                               0                            0  ...         1   
2                               0                            0  ...         0   
3                               0                            0  ...         0   
4                               0                            0  ...         0   
..                            ...                          ...  ...       ...   
143                             0                            0  ...         0   
144                             0                            0  ...         1   
145                             0                            0  ...         0   
146                             0                            0  ...         0   
147                             0                            0  ...         0   

     weepiness  weight gain  welt  wheelchair bound  wheezing  withdraw  \
0            0            0     0                 0         0         0   
1            0            0     0                 0         0         0   
2            1            0     0                 0         0         0   
3            1            0     0                 0         0         0   
4            0            0     0                 0         0         0   
..         ...          ...   ...               ...       ...       ...   
143          0            0     0                 0         0         0   
144          0            0     0                 0         0         0   
145          1            0     0                 0         0         0   
146          0            0     0                 0         0         0   
147          0            0     0                 0         0         0   

     worry  yellow  sputum  yellow sputum  
0        0               0              0  
1        0               0              0  
2        1               0              0  
3        1               0              0  
4        0               0              0  
..     ...             ...            ...  
143      0               0              0  
1

In [61]:
df.columns[420:450]

Index(['underweight', 'unhappy', 'unresponsiveness', 'unsteady  gait',
       'unsteady gait', 'unwell', 'urge  incontinence', 'urge incontinence',
       'urgency of micturition', 'urinary hesitation', 'urinoma',
       'verbal  auditory hallucinations', 'verbal auditory hallucinations',
       'verbally abusive behavior', 'vertigo', 'vision blurred', 'vomiting',
       'weepiness', 'weight gain', 'welt', 'wheelchair bound', 'wheezing',
       'withdraw', 'worry', 'yellow  sputum', 'yellow sputum'],
      dtype='object')

In [85]:
l1 = ["Heberden's node", "Murphy's sign", "Stahli's line", 'abdomen acute',
       'abdominal bloating', 'abdominal tenderness', 'abnormal sensation',
       'abnormally  hard consistency', 'abnormally hard consistency',
       'abortion', 'abscess bacterial', 'absences finding', 'achalasia',
       'ache', 'adverse effect', 'adverse reaction', 'agitation',
       'air fluid level', 'alcohol binge episode', 'alcoholic withdrawal symptoms', 'ambidexterity', 'angina  pectoris',
       'angina pectoris', 'anorexia', 'anosmia', 'aphagia', 'apyrexial',
       'arthralgia', 'ascites', 'asterixis', 'asthenia', 'asymptomatic',
       'ataxia', 'atypia', 'aura', 'awakening early', 'barking cough',
       'bedridden', 'behavior hyperactive', 'behavior showing increased motor activity', 'blackout', 'blanch',
       'bleeding of vagina', 'bowel sounds decreased', 'bradycardia',
       'bradykinesia', 'breakthrough pain', 'breath  sounds decreased',
       'breath sounds decreased', 'breath-holding spell',
       'breech presentation', 'bruit', 'burning  sensation',
       'burning sensation', 'cachexia', 'cardiomegaly',
       'cardiovascular  finding', 'cardiovascular event', 'catatonia', 'catching breath', 'charleyhorse', 'chest  tightness',
       'chest discomfort', 'chest tightness', 'chill', 'choke',
       'cicatrisation', 'clammy skin', 'claudication', 'clonus', 'clumsiness',
       'colic abdominal', 'consciousness clear', 'constipation',
       'coordination abnormal', 'cough', 'cushingoid facies',
       'cushingoid habitus', 'cyanosis', 'cystic lesion', 'debilitation', 'decompensation',
       'decreased  translucency', 'decreased body weight',
       'decreased stool caliber', 'decreased translucency', 'diarrhea',
       'difficulty', 'difficulty passing urine', 'disequilibrium',
       'distended  abdomen', 'distended abdomen', 'distress  respiratory',
       'distress respiratory', 'disturbed family', 'dizziness', 'dizzy spells',
       'drool', 'drowsiness', 'dullness', 'dysarthria', 'dysdiadochokinesia', 'dysesthesia',
       'dyspareunia', 'dyspnea', 'dyspnea  on exertion', 'dyspnea on exertion',
       'dysuria', 'ecchymosis', 'egophony', 'elation', 'emphysematous change',
       'energy  increased', 'energy increased', 'enuresis', 'erythema',
       'estrogen use', 'excruciating pain', 'exhaustion', 'extrapyramidal sign', 'extreme exhaustion', 'facial  paresis',
       'facial paresis', 'fall', 'fatigability', 'fatigue', 'fear of falling',
       'fecaluria', 'feces in rectum', 'feeling  hopeless',
       'feeling  suicidal', 'feeling hopeless', 'feeling strange',
       'feeling suicidal', 'feels hot/feverish', 'fever', 'flare',
       'flatulence', 'floppy', 'flushing', 'focal seizures', 'food  intolerance', 'food intolerance',
       'formication', 'frail', 'fremitus', 'frothy sputum', 'gag',
       'gasping for breath', 'general  discomfort', 'general discomfort',
       'general unsteadiness', 'giddy mood', 'gravida 0', 'gravida 10',
       'green  sputum', 'green sputum', 'groggy', 'guaiac positive', 'gurgle', 'hacking cough', 'haemoptysis', 'haemorrhage',
       'hallucinations  auditory', 'hallucinations  visual',
       'hallucinations auditory', 'hallucinations visual',
       'has religious belief', 'headache', 'heartburn', 'heavy feeling',
       'heavy legs', 'hematochezia', 'hematocrit decreased', 'hematuria',
       'heme positive', 'hemianopsia homonymous', 'hemiplegia',
       'hemodynamically stable', 'hepatomegaly', 'hepatosplenomegaly', 'hirsutism',
       'history  of - blackout', 'hoard', 'hoarseness', 'homelessness',
       'homicidal thoughts', 'hot flush', 'hunger', 'hydropneumothorax',
       'hyperacusis', 'hypercapnia', 'hyperemesis', 'hyperhidrosis disorder',
       'hyperkalemia', 'hypersomnia', 'hypersomnolence', 'hypertonicity',
       'hyperventilation', 'hypesthesia', 'hypoalbuminemia', 'hypocalcemia result', 'hypokalemia',
       'hypokinesia', 'hypometabolism', 'hyponatremia', 'hypoproteinemia',
       'hypotension', 'hypothermia, natural', 'hypotonic', 'hypoxemia',
       'immobile', 'impaired cognition', 'inappropriate affect', 'incoherent',
       'indifferent mood', 'intermenstrual heavy bleeding', 'intoxication',
       'irritable  mood', 'irritable mood', 'jugular  venous distention',
       'jugular venous distention', 'labored breathing', 'lameness',
       'large-for-dates fetus', 'left atrial hypertrophy', 'lesion',
       'lethargy', 'lightheadedness', 'lip smacking', 'loose associations',
       'low back pain', 'lung nodule', 'macerated skin', 'macule', 'malaise',
       'mass in breast', 'mass of body structure', 'mediastinal shift', 'mental  status changes', 'mental status changes', 'metastatic lesion',
       'milky', 'moan', 'monoclonal', 'monocytosis', 'mood  depressed',
       'mood depressed', 'moody', 'motor  retardation', 'motor retardation',
       'muscle hypotonia', 'muscle twitch', 'myalgia', 'mydriasis',
       'myoclonus', 'nasal discharge present', 'nasal flaring', 'nausea', 'nausea and vomiting', 'neck stiffness', 'neologism', 'nervousness',
       'night  sweat', 'night sweat', 'nightmare', 'no known drug allergies',
       'no status change', 'noisy respiration', 'non-productive  cough',
       'non-productive cough', 'nonsmoker', 'numbness', 'numbness  of hand',
       'numbness of hand', 'oliguria', 'orthopnea', 'orthostasis',
       'out  of breath', 'out of breath', 'overweight', 'pain', 'pain  chest', 'pain abdominal',
       'pain back', 'pain chest', 'pain foot', 'pain in lower limb',
       'pain neck', 'painful swallowing', 'pallor', 'palpitation', 'panic',
       'pansystolic murmur', 'para 1', 'para 2', 'paralyse', 'paraparesis',
       'paresis', 'paresthesia', 'passed stones', 'patient non compliance',
       'pericardial friction rub', 'phonophobia', 'photophobia', 'photopsia',
       'pin-point pupils', 'pleuritic  pain', 'pleuritic pain', 'pneumatouria',
       'polydypsia', 'polymyalgia', 'polyuria', 'poor dentition',
       'poor feeding', 'posterior rhinorrhea', 'posturing',
       'presence of q wave', 'pressure  chest', 'pressure chest', 'previous pregnancies 2', 'primigravida', 'prodrome',
       'productive  cough', 'productive cough', 'projectile vomiting',
       'prostate tender', 'prostatism', 'proteinemia', 'pruritus',
       'pulse absent', 'pulsus paradoxus', 'pustule', 'qt interval prolonged',
       'r wave feature', 'rale', 'rambling speech', 'rapid shallow breathing',
       'red blotches', 'redness', 'regurgitates after swallowing', 'renal angle tenderness',
       'rest pain', 'retch', 'retropulsion', 'rhd positive', 'rhonchus',
       'rigor - temperature-associated observation', 'rolling of eyes',
       'room spinning', 'satiety early', 'scar tissue', 'sciatica',
       'scleral icterus', 'scratch marks', 'sedentary', 'seizure',
       'sensory discomfort', 'shooting pain', 'shortness  of breath', 'shortness of breath', 'side pain',
       'sinus rhythm', 'sleeplessness', 'sleepy', 'slowing of urinary stream',
       'sneeze', 'sniffle', 'snore', 'snuffle', 'soft tissue swelling',
       'sore to touch', 'spasm', 'speech  slurred', 'speech slurred',
       'splenomegaly', 'spontaneous rupture of membranes', 'sputum purulent',
       'st segment depression', 'st segment elevation', 'stiffness', 'stinging sensation',
       'stool color yellow', 'stridor', 'stuffy nose', 'stupor', 'suicidal',
       'superimposition', 'sweat', 'sweating  increased', 'sweating increased',
       'swelling', 'symptom  aggravating factors',
       'symptom aggravating factors', 'syncope', 'systolic ejection murmur',
       'systolic murmur', 't wave inverted', 'tachypnea', 'tenesmus', 'terrify', 'thicken', 'throat sore',
       'throbbing sensation quality', 'tinnitus', 'tired', 'titubation',
       'todd paralysis', 'tonic seizures', 'transaminitis', 'transsexual',
       'tremor', 'tremor resting', 'tumor cell invasion',
       'unable  to concentrate', 'unable to concentrate', 'unconscious  state',
       'unconscious state', 'uncoordination', 'underweight', 'unhappy', 'unresponsiveness', 'unsteady  gait',
       'unsteady gait', 'unwell', 'urge  incontinence', 'urge incontinence',
       'urgency of micturition', 'urinary hesitation', 'urinoma',
       'verbal  auditory hallucinations', 'verbal auditory hallucinations',
       'verbally abusive behavior', 'vertigo', 'vision blurred', 'vomiting',
       'weepiness', 'weight gain', 'welt', 'wheelchair bound', 'wheezing',
       'withdraw', 'worry', 'yellow  sputum', 'yellow sputum']

In [86]:
def createList(r1, r2):
    return [item for item in range(r1, r2+1)]
     
# Driver Code
r1, r2 = 0, 450
l2 = createList(r1, r2)

In [87]:
Dic = {}
for key in l2:
    for value in l1:
        Dic[key] = value
        l1.remove(value)
        break
Dic

{0: "Heberden's node",
 1: "Murphy's sign",
 2: "Stahli's line",
 3: 'abdomen acute',
 4: 'abdominal bloating',
 5: 'abdominal tenderness',
 6: 'abnormal sensation',
 7: 'abnormally  hard consistency',
 8: 'abnormally hard consistency',
 9: 'abortion',
 10: 'abscess bacterial',
 11: 'absences finding',
 12: 'achalasia',
 13: 'ache',
 14: 'adverse effect',
 15: 'adverse reaction',
 16: 'agitation',
 17: 'air fluid level',
 18: 'alcohol binge episode',
 19: 'alcoholic withdrawal symptoms',
 20: 'ambidexterity',
 21: 'angina  pectoris',
 22: 'angina pectoris',
 23: 'anorexia',
 24: 'anosmia',
 25: 'aphagia',
 26: 'apyrexial',
 27: 'arthralgia',
 28: 'ascites',
 29: 'asterixis',
 30: 'asthenia',
 31: 'asymptomatic',
 32: 'ataxia',
 33: 'atypia',
 34: 'aura',
 35: 'awakening early',
 36: 'barking cough',
 37: 'bedridden',
 38: 'behavior hyperactive',
 39: 'behavior showing increased motor activity',
 40: 'blackout',
 41: 'blanch',
 42: 'bleeding of vagina',
 43: 'bowel sounds decreased',
 4

0: "Alzheimer's  disease"
1: "HIV"
2: "Pneumocystis  carinii pneumonia"
3: "accident  cerebrovascular"
4: "acquired  immuno-deficiency  syndrome"
5: "adenocarcinoma"
6: "adhesion"
7: "affect  labile"
8: "anemia"
9: "anxiety  state"
10: "aphasia"
11: "arthritis"
12: "asthma"
13: "bacteremia"
14: "benign  prostatic hypertrophy"
15: "biliary  calculus"
16: "bipolar  disorder"
17: "bronchitis"
18: "candidiasis"
19: "carcinoma"
20: "carcinoma breast"
21: "carcinoma colon"
22: "carcinoma of lung"
23: "carcinoma prostate"
24: "cardiomyopathy"
25: "cellulitis"
26: "cholecystitis"
27: "cholelithiasis"
28: "chronic  alcoholic intoxication"
29: "chronic  kidney failure"
30: "chronic  obstructive airway disease"
31: "cirrhosis"
32: "colitis"
33: "confusion"
34: "coronary  arteriosclerosis"
35: "coronary heart disease"
36: "decubitus  ulcer"
37: "deep  vein thrombosis"
38: "degenerative  polyarthritis"
39: "deglutition  disorder"
40: "dehydration"
41: "delirium"
42: "delusion"
43: "dementia"
44: "dependence"
45: "depression  mental"
46: "depressive disorder"
47: "diabetes"
48: "diverticulitis"
49: "diverticulosis"
50: "edema  pulmonary"
51: "effusion  pericardial"
52: "embolism  pulmonary"
53: "emphysema  pulmonary"
54: "encephalopathy"
55: "endocarditis"
56: "epilepsy"
57: "exanthema"
58: "failure  heart"
59: "failure  heart congestive"
60: "failure  kidney"
61: "fibroid  tumor"
62: "gastritis"
63: "gastroenteritis"
64: "gastroesophageal  reflux disease"
65: "glaucoma"
66: "gout"
67: "hemiparesis"
68: "hemorrhoids"
69: "hepatitis"
70: "hepatitis  B"
71: "hepatitis  C"
72: "hernia"
73: "hernia  hiatal"
74: "hiv infections"
75: "hyperbilirubinemia"
76: "hypercholesterolemia"
77: "hyperglycemia"
78: "hyperlipidemia"
79: "hypertension  pulmonary"
80: "hypertensive  disease"
81: "hypoglycemia"
82: "hypothyroidism"
83: "ileus"
84: "incontinence"
85: "infection"
86: "infection  urinary tract"
87: "influenza"
88: "insufficiency  renal"
89: "ischemia"
90: "ketoacidosis  diabetic"
91: "kidney  disease"
92: "kidney  failure acute"
93: "lymphatic  diseases"
94: "lymphoma"
95: "malignant  neoplasm of breast"
96: "malignant  neoplasm of lung"
97: "malignant  neoplasm of prostate"
98: "malignant  neoplasms"
99: "malignant  tumor of colon"
100: "manic  disorder"
101: "melanoma"
102: "migraine  disorders"
103: "mitral  valve insufficiency"
104: "myocardial  infarction"
105: "neoplasm"
106: "neoplasm  metastasis"
107: "neuropathy"
108: "neutropenia"
109: "obesity"
110: "obesity  morbid"
111: "oral  candidiasis"
112: "osteomyelitis"
113: "osteoporosis"
114: "overload  fluid"
115: "pancreatitis"
116: "pancytopenia"
117: "paranoia"
118: "parkinson  disease"
119: "paroxysmal  dyspnea"
120: "pericardial effusion body substance"
121: "peripheral  vascular disease"
122: "personality  disorder"
123: "pneumonia"
124: "pneumonia  aspiration"
125: "pneumothorax"
126: "primary  carcinoma of the liver cells"
127: "primary malignant neoplasm"
128: "psychotic  disorder"
129: "pyelonephritis"
130: "respiratory  failure"
131: "schizophrenia"
132: "sepsis (invertebrate)"
133: "septicemia"
134: "sickle  cell anemia"
135: "spasm  bronchial"
136: "stenosis  aortic valve"
137: "suicide  attempt"
138: "systemic  infection"
139: "tachycardia  sinus"
140: "thrombocytopaenia"
141: "thrombus"
142: "tonic-clonic  epilepsy"
143: "tonic-clonic seizures"
144: "transient  ischemic attack"
145: "tricuspid  valve insufficiency" 
146: "ulcer  peptic"
147: "upper  respiratory infection"

0: "Heberden's node",
1: "Murphy's sign",
2: "Stahli's line",
3: 'abdomen acute',
4: 'abdominal bloating',
5: 'abdominal tenderness',
6: 'abnormal sensation',
7: 'abnormally  hard consistency',
8: 'abnormally hard consistency',
9: 'abortion',
10: 'abscess bacterial',
11: 'absences finding',
12: 'achalasia',
13: 'ache',
14: 'adverse effect',
15: 'adverse reaction',
16: 'agitation',
17: 'air fluid level',
18: 'alcohol binge episode',
19: 'alcoholic withdrawal symptoms',
20: 'ambidexterity',
21: 'angina  pectoris',
22: 'angina pectoris',
23: 'anorexia',
24: 'anosmia',
25: 'aphagia',
26: 'apyrexial',
27: 'arthralgia',
28: 'ascites',
29: 'asterixis',
30: 'asthenia',
31: 'asymptomatic',
32: 'ataxia',
33: 'atypia',
34: 'aura',
35: 'awakening early',
36: 'barking cough',
37: 'bedridden',
38: 'behavior hyperactive',
39: 'behavior showing increased motor activity',
40: 'blackout',
41: 'blanch',
42: 'bleeding of vagina',
43: 'bowel sounds decreased',
44: 'bradycardia',
45: 'bradykinesia',
46: 'breakthrough pain',
47: 'breath  sounds decreased',
48: 'breath sounds decreased',
49: 'breath-holding spell',
50: 'breech presentation',
51: 'bruit',
52: 'burning  sensation',
53: 'burning sensation',
54: 'cachexia',
55: 'cardiomegaly',
56: 'cardiovascular  finding',
57: 'cardiovascular event',
58: 'catatonia',
59: 'catching breath',
60: 'charleyhorse',
61: 'chest  tightness',
62: 'chest discomfort',
63: 'chest tightness',
64: 'chill',
65: 'choke',
66: 'cicatrisation',
67: 'clammy skin',
68: 'claudication',
69: 'clonus',
70: 'clumsiness',
71: 'colic abdominal',
72: 'consciousness clear',
73: 'constipation',
74: 'coordination abnormal',
75: 'cough',
76: 'cushingoid facies',
77: 'cushingoid habitus',
78: 'cyanosis',
79: 'cystic lesion',
80: 'debilitation',
81: 'decompensation',
82: 'decreased  translucency',
83: 'decreased body weight',
84: 'decreased stool caliber',
85: 'decreased translucency',
86: 'diarrhea',
87: 'difficulty',
88: 'difficulty passing urine',
89: 'disequilibrium',
90: 'distended  abdomen',
91: 'distended abdomen',
92: 'distress  respiratory',
93: 'distress respiratory',
94: 'disturbed family',
95: 'dizziness',
96: 'dizzy spells',
97: 'drool',
98: 'drowsiness',
99: 'dullness',
100: 'dysarthria',
101: 'dysdiadochokinesia',
102: 'dysesthesia',
103: 'dyspareunia',
104: 'dyspnea',
105: 'dyspnea  on exertion',
106: 'dyspnea on exertion',
107: 'dysuria',
108: 'ecchymosis',
109: 'egophony',
110: 'elation',
111: 'emphysematous change',
112: 'energy  increased',
113: 'energy increased',
114: 'enuresis',
115: 'erythema',
116: 'estrogen use',
117: 'excruciating pain',
118: 'exhaustion',
119: 'extrapyramidal sign',
120: 'extreme exhaustion',
121: 'facial  paresis',
122: 'facial paresis',
123: 'fall',
124: 'fatigability',
125: 'fatigue',
126: 'fear of falling',
127: 'fecaluria',
128: 'feces in rectum',
129: 'feeling  hopeless',
130: 'feeling  suicidal',
131: 'feeling hopeless',
132: 'feeling strange',
133: 'feeling suicidal',
134: 'feels hot/feverish',
135: 'fever',
136: 'flare',
137: 'flatulence',
138: 'floppy',
139: 'flushing',
140: 'focal seizures',
141: 'food  intolerance',
142: 'food intolerance',
143: 'formication',
144: 'frail',
145: 'fremitus',
146: 'frothy sputum',
147: 'gag',
148: 'gasping for breath',
149: 'general  discomfort',
150: 'general discomfort',
151: 'general unsteadiness',
152: 'giddy mood',
153: 'gravida 0',
154: 'gravida 10',
155: 'green  sputum',
156: 'green sputum',
157: 'groggy',
158: 'guaiac positive',
159: 'gurgle',
160: 'hacking cough',
161: 'haemoptysis',
162: 'haemorrhage',
163: 'hallucinations  auditory',
164: 'hallucinations  visual',
165: 'hallucinations auditory',
166: 'hallucinations visual',
167: 'has religious belief',
168: 'headache',
169: 'heartburn',
170: 'heavy feeling',
171: 'heavy legs',
172: 'hematochezia',
173: 'hematocrit decreased',
174: 'hematuria',
175: 'heme positive',
176: 'hemianopsia homonymous',
177: 'hemiplegia',
178: 'hemodynamically stable',
179: 'hepatomegaly',
180: 'hepatosplenomegaly',
181: 'hirsutism',
182: 'history  of - blackout',
183: 'hoard',
184: 'hoarseness',
185: 'homelessness',
186: 'homicidal thoughts',
187: 'hot flush',
188: 'hunger',
189: 'hydropneumothorax',
190: 'hyperacusis',
191: 'hypercapnia',
192: 'hyperemesis',
193: 'hyperhidrosis disorder',
194: 'hyperkalemia',
195: 'hypersomnia',
196: 'hypersomnolence',
197: 'hypertonicity',
198: 'hyperventilation',
199: 'hypesthesia',
200: 'hypoalbuminemia',
201: 'hypocalcemia result',
202: 'hypokalemia',
203: 'hypokinesia',
204: 'hypometabolism',
205: 'hyponatremia',
206: 'hypoproteinemia',
207: 'hypotension',
208: 'hypothermia, natural',
209: 'hypotonic',
210: 'hypoxemia',
211: 'immobile',
212: 'impaired cognition',
213: 'inappropriate affect',
214: 'incoherent',
215: 'indifferent mood',
216: 'intermenstrual heavy bleeding',
217: 'intoxication',
218: 'irritable  mood',
219: 'irritable mood',
220: 'jugular  venous distention',
221: 'jugular venous distention',
222: 'labored breathing',
223: 'lameness',
224: 'large-for-dates fetus',
225: 'left atrial hypertrophy',
226: 'lesion',
227: 'lethargy',
228: 'lightheadedness',
229: 'lip smacking',
230: 'loose associations',
231: 'low back pain',
232: 'lung nodule',
233: 'macerated skin',
234: 'macule',
235: 'malaise',
236: 'mass in breast',
237: 'mass of body structure',
238: 'mediastinal shift',
239: 'mental  status changes',
240: 'mental status changes',
241: 'metastatic lesion',
242: 'milky',
243: 'moan',
244: 'monoclonal',
245: 'monocytosis',
246: 'mood  depressed',
247: 'mood depressed',
248: 'moody',
249: 'motor  retardation',
250: 'motor retardation',
251: 'muscle hypotonia',
252: 'muscle twitch',
253: 'myalgia',
254: 'mydriasis',
255: 'myoclonus',
256: 'nasal discharge present',
257: 'nasal flaring',
258: 'nausea',
259: 'nausea and vomiting',
260: 'neck stiffness',
261: 'neologism',
262: 'nervousness',
263: 'night  sweat',
264: 'night sweat',
265: 'nightmare',
266: 'no known drug allergies',
267: 'no status change',
268: 'noisy respiration',
269: 'non-productive  cough',
270: 'non-productive cough',
271: 'nonsmoker',
272: 'numbness',
273: 'numbness  of hand',
274: 'numbness of hand',
275: 'oliguria',
276: 'orthopnea',
277: 'orthostasis',
278: 'out  of breath',
279: 'out of breath',
280: 'overweight',
281: 'pain',
282: 'pain  chest',
283: 'pain abdominal',
284: 'pain back',
285: 'pain chest',
286: 'pain foot',
287: 'pain in lower limb',
288: 'pain neck',
289: 'painful swallowing',
290: 'pallor',
291: 'palpitation',
292: 'panic',
293: 'pansystolic murmur',
294: 'para 1',
295: 'para 2',
296: 'paralyse',
297: 'paraparesis',
298: 'paresis',
299: 'paresthesia',
300: 'passed stones',
301: 'patient non compliance',
302: 'pericardial friction rub',
303: 'phonophobia',
304: 'photophobia',
305: 'photopsia',
306: 'pin-point pupils',
307: 'pleuritic  pain',
308: 'pleuritic pain',
309: 'pneumatouria',
310: 'polydypsia',
311: 'polymyalgia',
312: 'polyuria',
313: 'poor dentition',
314: 'poor feeding',
315: 'posterior rhinorrhea',
316: 'posturing',
317: 'presence of q wave',
318: 'pressure  chest',
319: 'pressure chest',
320: 'previous pregnancies 2',
321: 'primigravida',
322: 'prodrome',
323: 'productive  cough',
324: 'productive cough',
325: 'projectile vomiting',
326: 'prostate tender',
327: 'prostatism',
328: 'proteinemia',
329: 'pruritus',
330: 'pulse absent',
331: 'pulsus paradoxus',
332: 'pustule',
333: 'qt interval prolonged',
334: 'r wave feature',
335: 'rale',
336: 'rambling speech',
337: 'rapid shallow breathing',
338: 'red blotches',
339: 'redness',
340: 'regurgitates after swallowing',
341: 'renal angle tenderness',
342: 'rest pain',
343: 'retch',
344: 'retropulsion',
345: 'rhd positive',
346: 'rhonchus',
347: 'rigor - temperature-associated observation',
348: 'rolling of eyes',
349: 'room spinning',
350: 'satiety early',
351: 'scar tissue',
352: 'sciatica',
353: 'scleral icterus',
354: 'scratch marks',
355: 'sedentary',
356: 'seizure',
357: 'sensory discomfort',
358: 'shooting pain',
359: 'shortness  of breath',
360: 'shortness of breath',
361: 'side pain',
362: 'sinus rhythm',
363: 'sleeplessness',
364: 'sleepy',
365: 'slowing of urinary stream',
366: 'sneeze',
367: 'sniffle',
368: 'snore',
369: 'snuffle',
370: 'soft tissue swelling',
371: 'sore to touch',
372: 'spasm',
373: 'speech  slurred',
374: 'speech slurred',
375: 'splenomegaly',
376: 'spontaneous rupture of membranes',
377: 'sputum purulent',
378: 'st segment depression',
379: 'st segment elevation',
380: 'stiffness',
381: 'stinging sensation',
382: 'stool color yellow',
383: 'stridor',
384: 'stuffy nose',
385: 'stupor',
386: 'suicidal',
387: 'superimposition',
388: 'sweat',
389: 'sweating  increased',
390: 'sweating increased',
391: 'swelling',
392: 'symptom  aggravating factors',
393: 'symptom aggravating factors',
394: 'syncope',
395: 'systolic ejection murmur',
396: 'systolic murmur',
397: 't wave inverted',
398: 'tachypnea',
399: 'tenesmus',
400: 'terrify',
401: 'thicken',
402: 'throat sore',
403: 'throbbing sensation quality',
404: 'tinnitus',
405: 'tired',
406: 'titubation',
407: 'todd paralysis',
408: 'tonic seizures',
409: 'transaminitis',
410: 'transsexual',
411: 'tremor',
412: 'tremor resting',
413: 'tumor cell invasion',
414: 'unable  to concentrate',
415: 'unable to concentrate',
416: 'unconscious  state',
417: 'unconscious state',
418: 'uncoordination',
419: 'underweight',
420: 'unhappy',
421: 'unresponsiveness',
422: 'unsteady  gait',
423: 'unsteady gait',
424: 'unwell',
425: 'urge  incontinence',
426: 'urge incontinence',
427: 'urgency of micturition',
428: 'urinary hesitation',
429: 'urinoma',
430: 'verbal  auditory hallucinations',
431: 'verbal auditory hallucinations',
432: 'verbally abusive behavior',
433: 'vertigo',
434: 'vision blurred',
435: 'vomiting',
436: 'weepiness',
437: 'weight gain',
438: 'welt',
439: 'wheelchair bound',
440: 'wheezing',
441: 'withdraw',
442: 'worry',
443: 'yellow  sputum',
444: 'yellow sputum'

**Spliting the data**

In [36]:
x = df.iloc[:, 1:].values
y = df.iloc[:, 0].values

**Random Forest Classifier**

In [51]:
RFA = RandomForestClassifier().fit(x, y)

**Save Model**

In [43]:
import pickle

In [44]:
# Save the model to a file using pickle
with open('RandomForest_model_148.pkl', 'wb') as model_file:
    pickle.dump(RFA, model_file)

--------------------------

Load the Model and test it in random inputs

In [48]:
with open('RandomForest_model_148.pkl', 'rb') as model_file:
    loaded_model = pickle.load(model_file)

In [50]:
preds = loaded_model.predict(np.array([[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0,0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0,0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0,0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0,0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,0, 0, 0, 0, 0]]))
print(preds[0])

80
